internal/util/util.go

```diff
@@ -74,33 +77,37 @@ func RequeueClusterOperator(c client.Client, name string) handler.MapFunc {
 	}
 }
 
-type POStatusErrors struct {
-	FailingPOs    []*platformv1alpha1.PlatformOperator
-	FailingErrors []error
-}
-
 // InspectPlatformOperators iterates over all the POs on the cluster
 // and determines whether a PO is in a failing state by inspecting its status.
 // A nil return value indicates no errors were found with the POs provided.
-func InspectPlatformOperators(POList *platformv1alpha1.PlatformOperatorList) *POStatusErrors {
-	POstatuses := new(POStatusErrors)
-
+func InspectPlatformOperators(POList *platformv1alpha1.PlatformOperatorList) error {
+	var failingPOs []error
 	for _, po := range POList.Items {
-		po := po.DeepCopy()
-		status := po.Status
-
-		for _, condition := range status.Conditions {
-			if condition.Reason == platformtypes.ReasonApplyFailed {
-				POstatuses.FailingPOs = append(POstatuses.FailingPOs, po)
-				POstatuses.FailingErrors = append(POstatuses.FailingErrors, fmt.Errorf("%s is failing: %q", po.GetName(), condition.Reason))
-			}
+		if err := inspectPlatformOperator(po); err != nil {
+			failingPOs = append(failingPOs, err)
 		}
 	}
-
-	// check if any POs were populated in the POStatusErrors type
-	if len(POstatuses.FailingPOs) > 0 || len(POstatuses.FailingErrors) > 0 {
-		return POstatuses
+	if len(failingPOs) > 0 {
+		return utilerror.NewAggregate(failingPOs)
 	}
+	return nil
+}
 
+// inspectPlatformOperator is responsible for inspecting an individual platform
+// operator resource, and determining whether it's reporting any failing conditions.
+// In the case that the PO resource is expressing failing states, then an error
+// will be returned to reflect that.
+func inspectPlatformOperator(po platformv1alpha1.PlatformOperator) error {
+	applied := meta.FindStatusCondition(po.Status.Conditions, platformtypes.TypeApplied)
+	if applied == nil {
+		return buildPOFailureMessage(po.GetName(), platformtypes.ReasonApplyPending)
+	}
```

Comment:
I _think_ this is the correct behavior but I'm open to other's thoughts here. Basically if there's no TypeApplied status condition type on the PO resource we're inspecting, then we can make the assumption that either the CPOM controller hasn't reconciled it yet, or there's a core issue in the CPOM controller that prevents that resource from being inspected.